File: AILang/benchmarks/python/format_hex.py
def format_hex_bench(iterations: int) -> int:
    sink = 0
    i = 0
    while i < iterations:
        sink += len(hex(i).upper().replace("X", "x"))
        i += 1
    return sink


def main() -> int:
    iterations = 400000
    result = format_hex_bench(iterations)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: AILang/benchmarks/rust/format_hex.rs
fn format_hex_bench(iterations: i64) -> i64 {
    let mut sink: i64 = 0;
    let mut i: i64 = 0;
    while i < iterations {
        sink += format!("0x{:X}", i).len() as i64;
        i += 1;
    }
    sink
}

fn main() {
    let iterations: i64 = 400_000;
    let result = format_hex_bench(iterations);
    println!("{}", result);
}
